packages/xstate-immer/README.md

```diff
@@ -1,3 +1,321 @@
+// { count: 2, level: 9 }
+
+toggleService.send(levelUpdater.update(-100));
+// Notice how the level is not updated in 'inactive' state:
+// { count: 2, level: 9 }
+```
+
+## API
+
+### `assign(producer)`
+
+Returns an XState event object that will update the machine's `context` to reflect the changes ("mutations") to `context` made in the `recipe` function.
+
+The `recipe` is similar to the function that you would pass to [Immer's `produce(val, recipe)` function](https://immerjs.github.io/immer/docs/produce)), with the addition that you get the same arguments as a normal XState assigner passed to `assign(assigner)` (`context`, `event`, `meta`).
+
+**Arguments for `assign`:**
+
+| Argument | Type     | Description                                                                                                             |
+| -------- | -------- | ----------------------------------------------------------------------------------------------------------------------- |
+| `recipe` | function | A function where "mutations" to `context` are made. See the [Immer docs](https://immerjs.github.io/immer/docs/produce). |
+
+**Arguments for `recipe`:**
+
+| Argument  | Type               | Description                                                              |
+| --------- | ------------------ | ------------------------------------------------------------------------ |
+| `context` | any                | The context data of the current state                                    |
+| `event`   | event object       | The received event object                                                |
+| `meta`    | assign meta object | An object containing meta data such as the `state`, SCXML `_event`, etc. |
+
+```js
+import { createMachine } from 'xstate';
+import { assign } from '@xstate/immer';
+
+const userMachine = createMachine({
+  id: 'user',
+  context: {
+    name: null,
+    address: {
+      city: null,
+      state: null,
+      country: null
+    }
+  },
+  initial: 'active',
+  states: {
+    active: {
+      on: {
+        CHANGE_COUNTRY: {
+          actions: assign((context, event) => {
+            context.address.country = event.value;
+          })
+        }
+      }
+    }
+  }
+});
+
+const { initialState } = userMachine;
+
+const nextState = userMachine.transition(initialState, {
+  type: 'UPDATE_COUNTRY',
+  country: 'USA'
+});
+
+nextState.context.address.country;
+// => 'USA'
+```
+
+### `createUpdater(eventType, recipe)`
+
+Returns an object that is useful for creating `context` updaters.
+
+| Argument    | Type     | Description                                                                       |
+| ----------- | -------- | --------------------------------------------------------------------------------- |
+| `eventType` | string   | The event type for the update event                                               |
+| `recipe`    | function | A function that takes in the `context` and some `input` to "mutate" the `context` |
```

Comment:
In the code this is called a `producer` - would be good to align the naming, probably requires a change in the code and not here, but it's up to you.

Reply:
Actually - I've just changed both of those names directly on your branch: https://github.com/davidkpiano/xstate/pull/1129/commits/5dc587a294260fb90fa7e17d6c2d6e62a9148baf

Feel free to adjust however you want